x-pack/plugins/data_visualizer/public/application/index_data_visualizer/search_strategy/requests/build_random_sampler_agg.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+
+/**
+ * Wraps the supplied aggregations in a random sampler aggregation.
+ */
+export function buildRandomSamplerAggregation(
+  aggs: any,
+  probability: number | null,
+  seed: number
+): Record<string, estypes.AggregationsAggregationContainer> {
```

Comment:
It looks like there are a couple of version of this type:
https://github.com/qn895/kibana/blob/75b2944216cda3f30dffc048972401a5e65e0af2/x-pack/plugins/data_visualizer/common/types/field_stats.ts#L238

https://github.com/qn895/kibana/blob/75b2944216cda3f30dffc048972401a5e65e0af2/x-pack/plugins/data_visualizer/common/utils/datafeed_utils.ts#L11

IMO the `type Aggregation = Record<string, estypes.AggregationsAggregationContainer>;` is better and matches how aggregations are described in the es client types.

If we clean these up and chose one type, this function could return `Record<Aggregation>`